packages/@react-aria/utils/src/isElementVisible.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2021 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+function isStyleVisible(element: Element) {
+  if (!(element instanceof HTMLElement) && !(element instanceof SVGElement)) {
+    return false;
+  }
+
+  let {display, visibility} = element.style;
+
+  let isVisible = (
+    display !== 'none' &&
+    visibility !== 'hidden' &&
+    visibility !== 'collapse'
+  );
+
+  if (isVisible) {
+    const {getComputedStyle} = element.ownerDocument.defaultView;
+    let {display: computedDisplay, visibility: computedVisibility} = getComputedStyle(element);
+
+    isVisible = (
+      computedDisplay !== 'none' &&
+      computedVisibility !== 'hidden' &&
+      computedVisibility !== 'collapse'
+    );
+  }
+
+  return isVisible;
+}
+
+function isAttributeVisible(element: Element, childElement?: Element) {
+  return (
+    !element.hasAttribute('hidden') &&
+    (element.nodeName === 'DETAILS' &&
+      childElement &&
+      childElement.nodeName !== 'SUMMARY'
+      ? element.hasAttribute('open')
+      : true)
+  );
+}
```

Comment:
I know this was sourced from jest-dom and that our selector filtering/isStyleVisible will ensure that elements with `hidden` aren't included but its kinda weird that this resolves to true if the element has `hidden` as an attribute

Reply:
Wouldn't that case be covered by `!element.hasAttribute('hidden') &&` at line 42?

Reply:
That then resolves to `false` for a element w/ `hidden` which then means the ternary returns `true` right?

Reply:
I think `isAttributeVisible` will return `false` for `!element.hasAttribute('hidden')` when the element has a `hidden` attribute before ever hitting the ternary, wrapped in parentheses, that follows: `(element.nodeName === 'DETAILS' && childElement && childElement.nodeName !== 'SUMMARY'  ? element.hasAttribute('open') : true)`. The ternary accounts for the edge case of a `<details>` without a `<summary>` that does not have an `open` attribute, the contents of which will not be visible. 

Example: https://codepen.io/majornista/pen/YzpeKRG?editors=1111